### DISCUSSION TITLE
Catching error message

### DISCUSSION THREAD

**Discussion by Akshaykakkodi** [pocketbase/pocketbase#4761] [Q&A] [answered]
Im creating a user in pocketbase with flutter app. Now when error happens while creating account, such as email already in use, invalid format, ect, how can i extract the error message from error object. im using try catch method for error handling.  

try{
await pb.collection(users).create(data:data);
}catch(e){
print(e);
}

i want to extract the error message in this object and display in ui.

**Comment by ganigeorgiev** [comment, marked as answer]
All Dart API requests throws a normalized `ClientException`. For more details about the available fields, please check [Error handling](https://github.com/pocketbase/dart-sdk?tab=readme-ov-file#error-handling).

You can use type assertion, ex. `e as ClientException`, variable or catch the specific error with the `on` clause, ex. `} on ClientException catch (e) {`.
